lms/models/solutions.py

```diff
@@ -46,6 +46,7 @@ def get_files_tree(files: Iterable[SolutionFile]) -> List[Dict[str, Any]]:
     file_details = [
         {
             'id': file.id,
+            'comments_num': len(Comment.by_file(file.id)),
```

Comment:
This is problematic. It will initiate lots and lots of queries in a short period of time.

We must find a way to retrieve in in a single query and to reduce the load on the database.